models/flexload_milp.py

```diff
@@ -293,9 +293,21 @@ def add_flex_tariffs(self, model, resolution="1h"):
             desired_charge_type="demand",
             model=model,
         )
-        model.total_flex_cost_signal = (
-            demand_flex_cost_signal + energy_flex_cost_signal
-        )
+        # when there is only one component in the cost function
+        # there is a RuntimeError due to identical components in the same block
+        try:
+            model.total_flex_cost_signal = (
+                energy_flex_cost_signal + demand_flex_cost_signal
+            )
+        except RuntimeError:
```

Comment:
Does it make sense to catch the specific `RuntimeError` or subclass of error here? Otherwise it looks good to me.